lib/ret_web/channels/hub_channel.ex

```diff
@@ -402,6 +617,121 @@ defmodule RetWeb.HubChannel do
 
   def handle_out("mute", _payload, socket), do: {:noreply, socket}
 
+  defp authorize_or_sanitize_update(payload, hub, sender_account, socket) do
+    %{"networkId" => network_id} = payload["data"]
+    secure_scene_object = socket.assigns.secure_scene_objects |> Enum.find(&(&1.network_id == network_id))
+    secure_template = secure_scene_object.template
+
+    if secure_template |> String.ends_with?("-media") do
+      is_creator = secure_scene_object.creator == socket.assigns.session_id
+      is_pinned = Repo.get_by(RoomObject, object_id: secure_scene_object.network_id) != nil
+
+      authorized =
+        (!is_pinned or sender_account |> can?(pin_objects(hub))) and
+          (is_creator or sender_account |> can?(spawn_and_move_media(hub)))
+
+      if authorized do
+        payload
+      else
+        payload |> Map.put("data", payload["data"] |> sanitize_with_authorized_schemas(secure_template, socket))
+      end
+    else
+      payload
+    end
+  end
+
+  defp maybe_store_secure_scene_object(socket, payload) do
+    %{"data" => %{"isFirstSync" => true, "creator" => creator, "template" => template, "networkId" => network_id}} =
+      payload
+
+    secure_scene_object = socket.assigns.secure_scene_objects |> Enum.find(&(&1.network_id == network_id))
+
+    if secure_scene_object == nil do
+      socket
+      |> assign(:secure_scene_objects, [
+        %{creator: creator, network_id: network_id, template: template} | socket.assigns.secure_scene_objects
+      ])
+    else
+      socket
+    end
+  end
+
+  defp authorize_object_manipulation(%{"networkId" => network_id} = data, socket) do
+    account = Guardian.Phoenix.Socket.current_resource(socket)
+    hub = socket |> hub_for_socket
```

Comment:
this is going to be problematic too, since we are going to be fetching from the db for every NAF message